packages/project/src/db/tree.ts

```diff
@@ -1,12 +1,113 @@
-import { type Tree, Instance } from "@webstudio-is/react-sdk";
+import { formatAsset } from "@webstudio-is/asset-uploader/server";
+import { type Tree, Instance, Text } from "@webstudio-is/react-sdk";
 import { applyPatches, type Patch } from "immer";
 import {
+  Asset,
   prisma,
   type Prisma,
   type Tree as DbTree,
 } from "@webstudio-is/prisma-client";
 import { utils } from "../index";
-import type { Breakpoint } from "@webstudio-is/css-data";
+import {
+  type Breakpoint,
+  SharedStyleValue,
+  ImageValue,
+} from "@webstudio-is/css-data";
+import { z } from "zod";
+import DataLoader from "dataloader";
+import warnOnce from "warn-once";
+
+const assetsLoader = new DataLoader<string, Asset | undefined>(
+  async (assetIds) => {
+    const assets = await prisma.asset.findMany({
+      where: {
+        id: {
+          in: [...assetIds],
+        },
+      },
+    });
+
+    return assetIds.map((assetId) =>
+      assets.find((asset) => asset.id === assetId)
+    );
```

Comment:
this is weird, looks like you are trying to ensure you only return assets that correspond the assetIds, but  how is it even possible that assets from the database would be anything but those assetIds from the query?

Reply:
Order after DB is not preserved and can be not the same as in `where.in`.
Dataloader needs the order of loaded items is the same as the keys order.

Reply:
please add a comment